packages/top-app-bar/index.tsx

```diff
@@ -19,28 +19,60 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-
-import React from 'react';
-import PropTypes from 'prop-types';
-import classnames from 'classnames';
-import TopAppBarFixedAdjust from './FixedAdjust';
+import * as React from 'react';
+import * as classnames from 'classnames';
+import TopAppBarFixedAdjust, {TopAppbarFixedAdjustProps} from './FixedAdjust';
 import {
   MDCFixedTopAppBarFoundation,
   MDCTopAppBarFoundation,
   MDCShortTopAppBarFoundation,
+// no mdc .d.ts file
+// @ts-ignore
 } from '@material/top-app-bar/dist/mdc.topAppBar';
 
-export default class TopAppBar extends React.Component {
-  foundation_ = null;
+export interface TopAppBarProps {
+  actionItems?: React.ReactElement<any>[];
+  className: string;
+  dense: boolean;
+  fixed: boolean;
+  navigationIcon?: React.ReactElement<any>;
+  prominent: boolean;
+  short: boolean;
+  shortCollapsed: boolean;
+  style: React.CSSProperties;
+  title: React.ReactNode;
+}
+
+interface TopAppBarState {
+  classList: Set<string>;
+  style: React.CSSProperties;
+}
+
+type Props = TopAppBarProps & React.HTMLProps<HTMLElement>;
+type VariantType = 'dense' | 'fixed' | 'prominent' | 'short' | 'shortCollapsed';
 
-  state = {
+export default class TopAppBar extends React.Component<
+  Props,
+  TopAppBarState
+  > {
+  topAppBarElement: React.RefObject<HTMLElement> = React.createRef();
+  foundation_?: MDCFixedTopAppBarFoundation | MDCTopAppBarFoundation | MDCShortTopAppBarFoundation;
+
```

Comment:
Is this union useful for a 3p developer? If so, consider making a type for MDCTopAppBarFoundation which is the union of these 3